clouddriver-aws/src/main/java/com/netflix/spinnaker/clouddriver/aws/provider/agent/AmazonInstanceTypeCachingAgent.java

```diff
@@ -103,155 +80,148 @@ public AmazonInstanceTypeCachingAgent(
+                    attributes.put("name", i.getInstanceType());
+                    attributes.put("defaultVCpus", i.getVCpuInfo().getDefaultVCpus());
+                    attributes.put("memoryInGiB", i.getMemoryInfo().getSizeInMiB() / 1024);
+                    attributes.put(
+                        "supportedArchitectures", i.getProcessorInfo().getSupportedArchitectures());
+
+                    if (i.getInstanceStorageInfo() != null) {
+                      InstanceStorageInfo info = i.getInstanceStorageInfo();
+                      Map<String, Object> instanceStorageAttributes = new HashMap<>();
+
+                      instanceStorageAttributes.put("totalSizeInGB", info.getTotalSizeInGB());
+                      if (info.getDisks() != null && info.getDisks().size() > 0) {
+                        instanceStorageAttributes.put(
+                            "storageTypes",
+                            info.getDisks().stream()
+                                .map(d -> d.getType())
+                                .collect(Collectors.joining(",")));
+                      }
+                      if (info.getNvmeSupport() != null) {
+                        instanceStorageAttributes.put("nvmeSupport", info.getNvmeSupport());
+                      }
+                      attributes.put("instanceStorageInfo", instanceStorageAttributes);
+                    }
+
+                    if (i.getGpuInfo() != null) {
+                      GpuInfo info = i.getGpuInfo();
+                      Map<String, Object> gpuInfoAttributes = new HashMap<>();
+
+                      if (info.getTotalGpuMemoryInMiB() != null) {
+                        gpuInfoAttributes.put("totalGpuMemoryInMiB", info.getTotalGpuMemoryInMiB());
+                      }
+                      if (info.getGpus() != null) {
+                        gpuInfoAttributes.put(
+                            "gpus",
+                            info.getGpus().stream()
+                                .map(
+                                    g -> {
+                                      Map<String, Object> gpuDeviceInfo = new HashMap<>();
+                                      gpuDeviceInfo.put("name", g.getName());
+                                      gpuDeviceInfo.put("manufacturer", g.getManufacturer());
+                                      gpuDeviceInfo.put("count", g.getCount());
+                                      gpuDeviceInfo.put(
+                                          "gpuSizeInMiB", g.getMemoryInfo().getSizeInMiB());
+                                      return gpuDeviceInfo;
+                                    })
+                                .collect(Collectors.toList()));
+                      }
+                      attributes.put("gpuInfo", gpuInfoAttributes);
+                    }
+
+                    if (i.getNetworkInfo() != null) {
+                      attributes.put("ipv6Supported", i.getNetworkInfo().getIpv6Supported());
```

Comment:
Would it be better to have this attribute not be optional? It's always a bit clunky consuming these values from the map when you have to do something like `if (attributes.get("key") != null && attributes.get("key") == true)` or equivalent.

Reply:
The `i` - [InstanceTypeInfo](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_InstanceTypeInfo.html) here is a result object returned from the AWS API `DescribeInstanceTypes` and `networkInfo` is an optional parameter. Also, not all instance types support IPv6, hence, the null check. Hope this adds clarification.